instrumentation/servlet/servlet-3.0/javaagent-tests/src/test/java/io/opentelemetry/javaagent/instrumentation/servlet/v3_0/tomcat/TestServlet3.java

```diff
@@ -0,0 +1,284 @@
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class TestServlet3 {
+
+  private TestServlet3() {}
+
+  @WebServlet
+  public static class Sync extends HttpServlet {
+    @Override
+    protected void service(HttpServletRequest req, HttpServletResponse resp) {
+      String servletPath = (String) req.getAttribute(RequestDispatcher.INCLUDE_SERVLET_PATH);
+      if (servletPath == null) {
+        servletPath = req.getServletPath();
+      }
+
+      ServerEndpoint endpoint = ServerEndpoint.forPath(servletPath);
+      HttpServerTest.controller(
+          endpoint,
+          () -> {
+            resp.setContentType("text/plain");
+            if (SUCCESS.equals(endpoint)) {
+              resp.setStatus(endpoint.getStatus());
+              resp.getWriter().print(endpoint.getBody());
+            } else if (INDEXED_CHILD.equals(endpoint)) {
+              endpoint.collectSpanAttributes(req::getParameter);
+              resp.setStatus(endpoint.getStatus());
+            } else if (QUERY_PARAM.equals(endpoint)) {
+              resp.setStatus(endpoint.getStatus());
+              resp.getWriter().print(req.getQueryString());
+            } else if (REDIRECT.equals(endpoint)) {
+              resp.sendRedirect(endpoint.getBody());
+            } else if (CAPTURE_HEADERS.equals(endpoint)) {
+              resp.setHeader("X-Test-Response", req.getHeader("X-Test-Request"));
+              resp.setStatus(endpoint.getStatus());
+              resp.getWriter().print(endpoint.getBody());
+            } else if (CAPTURE_PARAMETERS.equals(endpoint)) {
+              req.setCharacterEncoding("UTF8");
+              String value = req.getParameter("test-parameter");
+              if (!value.equals("test value õäöü")) {
+                throw new ServletException(
+                    "request parameter does not have expected value " + value);
+              }
+
+              resp.setStatus(endpoint.getStatus());
+              resp.getWriter().print(endpoint.getBody());
+            } else if (ERROR.equals(endpoint)) {
+              resp.sendError(endpoint.getStatus(), endpoint.getBody());
+            } else if (EXCEPTION.equals(endpoint)) {
+              throw new ServletException(endpoint.getBody());
+            } else if (HTML_PRINT_WRITER.equals(endpoint)) {
+              resp.setContentType("text/html");
+              resp.setStatus(endpoint.getStatus());
+              resp.setContentLength(endpoint.getBody().length());
+              resp.getWriter().print(endpoint.getBody());
+            } else if (HTML_SERVLET_OUTPUT_STREAM.equals(endpoint)) {
+              resp.setContentType("text/html");
+              resp.setStatus(endpoint.getStatus());
+              resp.setContentLength(endpoint.getBody().length());
+              resp.getOutputStream().print(endpoint.getBody());
```

Comment:
This alters the behavior of the test. Before it tested the `write(byte b[], int off, int len)` method now as far as I can tell it tests the `write(int b)` method. Was this change intentional? Did you check that we don't loose test coverage with this change? I guess this change just changes which method we test. Idk if we have some other test that would covers both of these.

Reply:
I don't understand - this was the old code:

```
} else if (HTML_SERVLET_OUTPUT_STREAM.equals(endpoint)) {
              resp.setContentType("text/html");
              resp.setStatus(endpoint.getStatus());
              resp.setContentLength(endpoint.getBody().length());
              resp.getOutputStream().print(endpoint.getBody());
            }
```

where is the difference?

Reply:
Old code is
https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/98bf18370c2c81f1c76d1031f452b2da52776ff4/instrumentation/servlet/servlet-3.0/javaagent/src/test/groovy/TestServlet3.groovy#L80-L86